platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/DtoConverter.java

```diff
@@ -59,30 +48,37 @@
     private DtoConverter() {
     }
 
-    public static ProjectTypeDefinition toTypeDefinition(ProjectType projectType) {
-        final ProjectTypeDefinition definition = newDto(ProjectTypeDefinition.class).withId(projectType.getId())
-                                                                                    .withDisplayName(projectType.getDisplayName())
-                                                                                    .withPrimaryable(projectType.canBePrimary())
-                                                                                    .withMixable(projectType.canBeMixin());
+    public static ProjectTypeDto toTypeDefinition(ProjectTypeDef projectType) {
+        final ProjectTypeDto definition = newDto(ProjectTypeDto.class).withId(projectType.getId())
+                                                                      .withDisplayName(projectType.getDisplayName())
+                                                                      .withPrimaryable(projectType.isPrimaryable())
+                                                                      .withMixable(projectType.isMixable())
+                                                                      .withAncestors(projectType.getAncestors());
 
-        final List<AttributeDescriptor> typeAttributes = new ArrayList<>();
+        final List<AttributeDto> typeAttributes = new ArrayList<>();
         for (Attribute attr : projectType.getAttributes()) {
-            List<String> valueList = null;
+
+
+            ValueDto valueDto = newDto(ValueDto.class);
+
             if (attr.getValue() != null) {
-                valueList = attr.getValue().getList();
+                valueDto.withList(attr.getValue().getList());
             }
-            typeAttributes.add(newDto(AttributeDescriptor.class).withName(attr.getName())
-                                                                .withDescription(attr.getDescription())
-                                                                .withRequired(attr.isRequired())
-                                                                .withVariable(attr.isVariable())
-                                                                .withValues(valueList));
+
+            typeAttributes.add(newDto(AttributeDto.class).withName(attr.getName())
+                                                         .withDescription(attr.getDescription())
+                                                         .withRequired(attr.isRequired())
+                                                         .withVariable(attr.isVariable())
+                                                         .withValue(valueDto));
         }
-        definition.setAttributeDescriptors(typeAttributes);
+        definition.withAttributes(typeAttributes).withParents(projectType.getParents());
+
+//        final List<String> parents = projectType.getParents().stream()
```

Comment:
why do we have commented code, either it needs to be removed or add TODO, FIXME, etc